### PR TITLE
Tests, NDEBUG check removal, various fixes and etc.

### PR DESCRIPTION
**Pull request by rbx**
- Add CMake variable to control `FAIR_MIN_SEVERITY`.
- Fix file sink and update `fMinSeverity` on its removal.
- Move `Logging()` implementation to header (reduces the runtime of nologTest from 319ms to 68ms).
- Make `GetColoredSeverityString()` public.
- `InitFileSink()`: return name of the created file.
- Throw if removing non-existing sink or adding existing.
- Consider `FAIR_MIN_SEVERITY` when setting severities - required for proper function of `Logging()` and others.
- Remove `NDEBUG` check. Compile-time severity setting is now opt-in only.
- Add more unit tests: severities, verbosities, nolog performance, special macros, file/custom sinks, multi-threading, cycle methods.
- Add getters for file & custom sink severity.
- Add output operator for severities, verbosities.

### CODE REVIEW

**Comment by rbx** [comment]
~~Looks like MacOS has trouble with the regex string. Gonna check.~~ Fixed.

File: logger/Logger.cxx [hunk 1]
```diff
@@ -573,7 +605,16 @@ void Logger::AddCustomSink(const string& key, Severity severity, function<void(c
 {
     lock_guard<mutex> lock(fMtx);
     if (fCustomSinks.count(key) == 0) {
+#ifdef FAIR_MIN_SEVERITY
```
Comment (dennisklein) [inline logger/Logger.cxx]: Let's always set `FAIR_MIN_SEVERITY`.

Comment (rbx) [reply logger/Logger.cxx]: Done.

**Review by dennisklein — APPROVED** [review]

**Comment by rbx** [comment]
@ktf @davidrohr This PR removes the `NDEBUG` check. If you want `FAIR_MIN_SEVERITY` to depend on it, you have to check for it in your project and set the `FAIR_MIN_SEVERITY` to a desired value. I prefer to have this opt-in only.

**Comment by davidrohr** [comment]
@ktf : How do we do this best?
- Ideally I'd say in alidist/o2.sh, but would we make it depend on? CMAKE_BUILD_TYPE!=DEBUG?
- Could do it in O2/CMakeLists.txt and set FAIR_MIN_SEVERITY before FindPackage(FairLogger) when there is NDEBUG in the CXXFLAGS.
- Could also do it in Framework/Logger, but that will only work if people use Framework/Logger and not FairLogger directly.

**Pull request merged**